app/controllers/api/events_controller.rb

```diff
@@ -0,0 +1,36 @@
+class API::EventsController < ApplicationController
+  skip_before_action :verify_authenticity_token
+  before_filter :set_access_control_headers
+
+  def set_access_control_headers
+    headers['Access-Control-Allow-Origin'] = '*'
+    headers['Access-Control-Allow-Methods'] = 'POST, GET, OPTIONS'
+    headers['Access-Control-Allow-Headers'] = 'Content-Type'
+  end
+
+
+  def create
+    registered_application = RegisteredApplication.find_by(url: request.env['HTTP_ORIGIN'])
+    if registered_application.nil?
```

Comment:
You could also write this in one line and have the if statement following the json rendering.